pkg/util/logger_windows.go

```diff
@@ -0,0 +1,134 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/go-kit/kit/log"
+	"github.com/weaveworks/common/server"
+	el "golang.org/x/sys/windows/svc/eventlog"
+)
+
+// Default name for the Grafana Agent under Windows
+const ServiceName = "Grafana Agent"
+
+//NewWindowsEventLogger creates a new logger that writes to the event log
+func NewWindowsEventLogger(cfg *server.Config) *Logger {
+	l := Logger{makeLogger: makeWindowsEventLogger}
+	if err := l.ApplyConfig(cfg); err != nil {
+		panic(err)
+	}
+	return &l
+}
+
+func makeWindowsEventLogger(cfg *server.Config) (log.Logger, error) {
+	// error is necessary for the Log function to return an error
+	notAllowedError := errors.New("not_allowed")
+
+	// Setup the log in windows events
+	err := el.InstallAsEventCreate(ServiceName, el.Error|el.Info|el.Warning)
+
+	// We expect an error of 'already exists' for subsequent runs,
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return nil, err
+	}
+	il, err := el.Open(ServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cleanup the handle when exits scope, this could be handled via explicit close but would need to change
+	// more upstream, and its not a huge issue if it hangs around slightly longer than it should
+	runtime.SetFinalizer(il, func(l *el.Log) {
+		l.Close()
+	})
+
+	//  These are setup to be writers for each Windows log level
+	//  Setup this way so we can utilize all the benefits of logformatter
+	infoWriter := &winLogWriter{writer: func(p []byte) {
+		il.Info(1, string(p))
+	}}
+	infoLogger := log.NewLogfmtLogger(infoWriter)
+	infoLogger = level.NewFilter(infoLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	warningWriter := &winLogWriter{writer: func(p []byte) {
+		il.Warning(1, string(p))
+	}}
+	warningLogger := log.NewLogfmtLogger(warningWriter)
+	warningLogger = level.NewFilter(warningLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	errorWriter := &winLogWriter{writer: func(p []byte) {
+		il.Error(1, string(p))
+	}}
+	errorLogger := log.NewLogfmtLogger(errorWriter)
+	errorLogger = level.NewFilter(errorLogger, cfg.LogLevel.Gokit, level.ErrNotAllowed(notAllowedError))
+
+	wl := &winLoggerFmt{
+		internalLog:   il,
+		errorLogger:   errorLogger,
+		infoLogger:    infoLogger,
+		warningLogger: warningLogger,
+	}
+	return wl, nil
+
+}
+
+type winLoggerFmt struct {
+	internalLog   *el.Log
+	errorLogger   log.Logger
+	infoLogger    log.Logger
+	warningLogger log.Logger
+}
+
+func (w *winLoggerFmt) Log(keyvals ...interface{}) error {
+	lvl, err := getLevel(keyvals...)
+	// If we don't know what happened move on
+	if err != nil {
+		w.infoLogger.Log(keyvals...)
+		return err
+	}
```

Comment:
We can probably have `getLevel` return nil if it didn't find a level and rely on the default case of the switch statement to clean this function up.